### PR TITLE
added gitea provider

### PR DESCRIPTION
**Pull request by ederuiter**
This adds support for gitea (https://gitea.io). The code is largely copy/pasted from the gitlab provider, but with some minor changes (different url's + other fields returned from the user api)

### CODE REVIEW

**Comment by atymic** [comment]
@ederuiter 
Please PR docs. Will close in one 1 week if no activity.

**Comment by atymic** [comment]
Closing due to inactivity. Please ping me if you'd like to get this merged and I will repopen.

**Comment by ederuiter** [comment]
@atymic I have created PR for the website https://github.com/SocialiteProviders/website/pull/9 
Is this sufficient or do you need more?

PS: https://socialiteproviders.com/contribute/#creating-a-provider does not mention the docs PR

File: src/Gitea/composer.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "name": "socialiteproviders/gitea",
+    "description": "Gitea OAuth2 Provider for Laravel Socialite",
+    "license": "MIT",
+    "authors": [{
+        "name": "Eric de Ruiter",
+        "email": "eric@thisisdevelopment.nl"
+    }],
+    "require": {
+        "php": "^5.6 || ^7.0",
+        "socialiteproviders/manager": "~2.0 || ~3.0"
```
Comment (atymic) [inline src/Gitea/composer.json:11]: drop v2 here please

File: src/Gitea/composer.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "socialiteproviders/gitea",
+    "description": "Gitea OAuth2 Provider for Laravel Socialite",
+    "license": "MIT",
+    "authors": [{
+        "name": "Eric de Ruiter",
+        "email": "eric@thisisdevelopment.nl"
+    }],
+    "require": {
+        "php": "^5.6 || ^7.0",
```
Comment (atymic) [inline src/Gitea/composer.json:10]: drop < 7.2 please

**Review by atymic — CHANGES_REQUESTED** [review]

**Comment by atymic** [comment]
@ederuiter 
To generate the docs (now live in this repo) run the `doc-gen.php` script in the tools folder 👍

**Pull request merged**